cli/config.go

```diff
@@ -50,19 +86,116 @@ func ParseConfig() {
 		log.Fatal(err)
 	}
 
-	if config.Offset != 0 {
-		Offset = config.Offset
+	if config.Filter.Offset != "" {
+		offset, err := strconv.ParseUint(config.Filter.Offset, 10, 64)
+		if err != nil {
+			fmt.Println("Error converting Offset to uint64:", err)
+			return
+		}
+		Offset = offset
 	}
 
-	if config.Limit != 0 {
-		Limit = config.Limit
+	if config.Filter.Limit != "" {
+		limit, err := strconv.ParseUint(config.Filter.Limit, 10, 64)
+		if err != nil {
+			fmt.Println("Error converting Offset to uint64:", err)
+			return
+		}
+		Limit = limit
 	}
 
-	if config.Name != "" {
-		Name = config.Name
+	if config.Filter.Name != "" {
+		Name = config.Filter.Name
 	}
 
-	if config.RawOutput {
-		RawOutput = config.RawOutput
+	if config.Filter.RawOutput != "" {
+		rawOutput, err := strconv.ParseBool(config.Filter.RawOutput)
+
+		if err != nil {
+			fmt.Println("Error converting string to bool:", err)
+		}
+
+		RawOutput = rawOutput
 	}
+
+}
+
+func NewConfigCmd() *cobra.Command {
+	return &cobra.Command{
+		Use:   "config",
+		Short: "CLI local config",
+		Long:  "Local param storage to prevent repetitive passing of keys",
+		Run: func(cmd *cobra.Command, args []string) {
+			fmt.Println("Configuring CLI...")
+			var key, value string
+
+			// Prompt the user for inputs
+			fmt.Print("Enter key: ")
+			fmt.Scan(&key)
+
+			fmt.Print("Enter value: ")
+			fmt.Scan(&value)
+
+			setConfigValue(key, value)
+			fmt.Println("Configuration complete")
```

Comment:
this would be better done as part of the cli args rather than an interactive session

Reply:
Okay.
Is there any particular reason for it?

Reply:
yes this whole session can be broken down to a single command something like `cli set config <key> <value>`

Reply:
I have addressed the comments